action_server/src/robocorp/action_server/_server.py

```diff
@@ -104,5 +101,38 @@ async def serve_index(request: Request):
             include_in_schema=False,
         )
 
+    def check_port(loop):
+        if not server.started:
+            loop.call_later(1 / 15.0, partial(check_port, loop))
+            return
+
+        port = settings.port if settings.port != 0 else None
+        if port is None:
+            port = server.servers[0].sockets[0].getsockname()[1]
```

Comment:
There's a caveat here. You could have more than one socket, one ipv4 and one ipv6. In this case, you need to do a for and check which one is ipv4 (because the _server_expose.py can't deal with ipv6 in the current structure).

Reply:
https://github.com/robocorp/robo/pull/96/commits/9f24e8c3e52bfdfa2f094af9991f813cc2bb4d20 should do the trick 👍